### PR TITLE
[KYUUBI #3705] Add docs for JDBC authentication usage with in-memory database

### LINKED ISSUE #3705: [DOCS] Add docs for JDBC authentication usage with in-memory database
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Which parts of the documentation do you think need improvement?

Add docs for JDBC authentication usage with in-memory database.

Used with auto created in-memory database, JDBC authentication could be applied for token validation without starting up a dedicated database service or a custom plugin.

### Affects Version(s)

1.6.0

### Improving the documentation

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes. I can submit a PR independently to improve.
- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

to close #3705 .

Add docs for JDBC authentication usage with in-memory database with config example for token validation example.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
Followed with #3701 , and please consider merging to master and 1.6 .

File: docs/security/jdbc.md [hunk 1]
```diff
@@ -46,4 +46,19 @@ kyuubi.authentication.jdbc.url = jdbc:mysql://127.0.0.1:3306/auth_db
...
+Consider authenticating a pair of a username and a token which contacted with an `expire_time` in 'yyyyMMddHHmm' format and a MD5 signature generated with sequence of `expire_time`, `username` and a secret key. With the following example, an H2 in-memory database will be auto crated with Kyuubi Server and used for authentication with its system function `HASH` and checking token expire time with `NOW()`.
+
+```properties
+kyuubi.authentication=JDBC
+kyuubi.authentication.jdbc.driver.class = org.h2.Driver
+kyuubi.authentication.jdbc.url = jdbc:h2:mem:
+kyuubi.authentication.jdbc.user = no_user
+kyuubi.authentication.jdbc.query = SELECT 1 FROM (SELECT ${user} as username, 'secret_key' as secret_key, SUBSTRING(${password}, 0, 12) as expire_time, SUBSTRING(${password}, 13) as signed
+) WHERE signed = RAWTOHEX(HASH('MD5', CONCAT(secret_key, username, expire_time))) AND PARSEDATETIME(expire_time,'yyyyMMddHHmm') > NOW();
+```
```
Comment (pan3793) [inline docs/security/jdbc.md:64]: Kyuubi conf file support using back slash to set multi-lines value

Comment (bowenliang123) [reply docs/security/jdbc.md:68]: Thanks, updated as suggested.

**Comment by pan3793** [comment]
Thanks, merging to master/1.6

**Pull request closed without merging**